lib/widgets/action_sheet.dart

```diff
@@ -41,25 +44,50 @@ void showMessageActionSheet({required BuildContext context, required Message mes
       && reactionWithVotes.userIds.contains(store.selfUserId))
     ?? false;
 
-  showDraggableScrollableModalBottomSheet<void>(
+  final designVariables = DesignVariables.of(context);
+  showModalBottomSheet<void>(
     context: context,
+    clipBehavior: Clip.antiAlias,
+    backgroundColor: designVariables.actionSheetBackground,
+    shape: const RoundedRectangleBorder(borderRadius: BorderRadius.vertical(top: Radius.circular(20.0))),
+    useSafeArea: true,
+    isScrollControlled: true,
     builder: (BuildContext _) {
-      return Column(children: [
-        if (!hasThumbsUpReactionVote) AddThumbsUpButton(message: message, messageListContext: context),
-        StarButton(message: message, messageListContext: context),
-        if (isComposeBoxOffered) QuoteAndReplyButton(
-          message: message,
-          messageListContext: context,
-        ),
-        if (showMarkAsUnreadButton) MarkAsUnreadButton(
-          message: message,
-          messageListContext: context,
-          narrow: narrow,
+      return Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            // TODO(#217): show message text
+            Flexible(
+              child: SingleChildScrollView(
+                child: ClipRRect(
+                  borderRadius: BorderRadius.circular(7),
+                  child: Column(spacing: 1, children: [
+                    if (!hasThumbsUpReactionVote) AddThumbsUpButton(message: message, messageListContext: context),
+                    StarButton(message: message, messageListContext: context),
+                    if (isComposeBoxOffered) QuoteAndReplyButton(
+                      message: message,
+                      messageListContext: context,
+                    ),
+                    if (showMarkAsUnreadButton) MarkAsUnreadButton(
+                      message: message,
+                      messageListContext: context,
+                      narrow: narrow,
+                    ),
+                    CopyMessageTextButton(message: message, messageListContext: context),
+                    CopyMessageLinkButton(message: message, messageListContext: context),
+                    ShareButton(message: message, messageListContext: context),
+                  ]),
+                ),
+              ),
+            ),
+            const SizedBox(height: 8),
```

Comment:
When the buttons area can scroll (when there are more buttons than fit on the user's screen), the content, as it scrolls, should be allowed to show through this 8px gap, but masked with a gradient from transparent to `DesignVariables.bgContextMenu`: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3483-42553&m=dev

(Same with an area of height 8px above the buttons.)

If this turns out to be difficult, we can file an issue with a TODO pointing to it, but let's not lose track of it. It seems independent of #217. #217 is one factor that will make this scroll effect more noticeable (by taking up more vertical space), but it's not the only factor that can; other factors are the number of buttons, the user's screen size, and the user's system text-size setting.